Source/Particles/Collision/BinaryCollision/ProtonBoronFusionInitializeMomentum.H

```diff
@@ -22,35 +28,227 @@ namespace {
     AMREX_GPU_HOST_DEVICE AMREX_INLINE
     void ProtonBoronFusionInitializeMomentum (
-                            const SoaData_type /*soa_1*/, const SoaData_type /*soa_2*/,
-                            const SoaData_type soa_alpha,
-                            const index_type& /*idx_1*/, const index_type& /*idx_2*/,
+                            const SoaData_type& soa_1, const SoaData_type& soa_2,
+                            SoaData_type& soa_alpha,
+                            const index_type& idx_1, const index_type& idx_2,
                             const index_type& idx_alpha_start,
-                            const amrex::ParticleReal& /*m1*/, const amrex::ParticleReal& /*m2*/)
+                            const amrex::ParticleReal& m1, const amrex::ParticleReal& m2,
+                            const amrex::RandomEngine& engine)
     {
-        soa_alpha.m_rdata[PIdx::ux][idx_alpha_start] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::uy][idx_alpha_start] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::uz][idx_alpha_start] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::ux][idx_alpha_start + 1] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::uy][idx_alpha_start + 1] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::uz][idx_alpha_start + 1] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::ux][idx_alpha_start + 2] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::uy][idx_alpha_start + 2] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::uz][idx_alpha_start + 2] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::ux][idx_alpha_start + 3] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::uy][idx_alpha_start + 3] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::uz][idx_alpha_start + 3] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::ux][idx_alpha_start + 4] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::uy][idx_alpha_start + 4] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::uz][idx_alpha_start + 4] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::ux][idx_alpha_start + 5] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::uy][idx_alpha_start + 5] = amrex::ParticleReal(0.);
-        soa_alpha.m_rdata[PIdx::uz][idx_alpha_start + 5] = amrex::ParticleReal(0.);
+        // General notations in this function:
+        //     x_sq denotes the square of x
+        //     x_star denotes the value of x in the proton+boron center of mass frame
+        //     x_Bestar denotes the value of x in the Beryllium rest frame
+
+        using namespace amrex::literals;
+
+        constexpr amrex::ParticleReal mev_to_joule = PhysConst::q_e*1.e6_prt;
+        // Energy produced in the fusion reaction proton + boron11 -> Beryllium8 + alpha
+        // cf. Janis book of proton-induced cross-sections (2019)
+        constexpr amrex::ParticleReal E_fusion = 8.59009_prt*mev_to_joule;
```

Comment:
@NeilZaim By computing `(m_p + m_B11 - m_alpha - m_B8)*c**2` (using WarpX data for the masses), I found 8.0972 MeV, instead of 8.59009 MeV.
Is this due to some inconsistencies in the definition of the of the masses for hydrogen and boron 11?

Reply:
Ah, actually, this is addressed below in the comments.